docs/dev-guide/adding-grid-support/adding-grid-support-step-by-step-guide/add-grid-config.md

```diff
@@ -1,3 +1,175 @@
+
+When setting up a new grid you will need to edit some or all of these files:
+
+- `cime_config/config_grids.xml`
+- `components/eam/bld/config_files/horiz_grid.xml`
+- `components/eam/bld/namelist_files/namelist_defaults_eam.xml`
+- `components/eam/bld/namelist_files/namelist_definition.xml`
+- `components/elm/bld/namelist_files/namelist_definition.xml`
+- `components/elm/bld/namelist_files/namelist_defaults.xml`
+
+## Mono-Grid vs Bi-Grid vs Tri-Grid
+
+The mono-bi-tri grid options in E3SM can be confusing, but it's important to understand what these terms mean when adding a new grid to E3SM. At the surface these terms mean that the whole model either using a single grid for all componennt models, or a combination of 2 or 3 grids shared among the component models. Note that mono-grid and bi-grid terms often ignore that the river model needs to be on i/ts own regular lat-lon grid (for now).
+
+Historically, climate models would use a single grid for all components (i.e. mono-grid), but this is often not the case anymore. In E3SM the ocean and sea-ice components often use targeted regional refinement with special consideration of ocean mesoscale eddies, whereas the atmosphere will generally use a globally homogenous grid. In practice, the main difference between "bi" and "tri" grids often comes down to whether the land surface model shares a grid with the atmosphere or not. The component coupler is in charge of facilitating communication between component models, primarily through fluxes, and so mapping files are needed to support a combination of different grids. E3SMv3 uses a tri-grid configuration for production simulations.
+
+## Grid Naming Conventions
+
+The atmosphere grid name should always indicate the "ne" value and add "pg2" to indicate that the physgrid is being used. For a regionally refined mesh (RRM) the grid name should always start with `ne0` followed a descriptive string that includes the region being refined and the degree of refinement.
+
+**Example**: `ne0np4_northamericax4v1`
+
+Note that the example indicates a `4x` refinement, but does indicate the base resolution, which is useful to know. A better grid name would be `ne0np4_northamerica30x4v1`, because this tells us that the grid is consistent with `ne30` in the unrefined regions.
+
+For a rectilinear lat-lon grid used by the land and/or river models the grid name should start with "r" and typically use spacing less than one degree, so they indicate the nominal grid spacing, starting with "0" and omitting the decimal.
+
+**Examples**: `r05` is 0.5 degree spacing and `r0125` is 1/8 or 0.125 degree spacing.
+
+For a mono-grid, which can only be used for idealized simulations such as aqua planet and RCE, the grid is written twice to indicate that both atmosphere/land and ocean/sea-ice models are on the same grid.
+
+**Example**: `ne30pg2_ne30pg2`
+
+Bi-grid options should indicate two different grids used for atmosphere/land and ocean/sea-ice models.
```

Comment:
A general point: if you're going to go through the trouble of trying to explain the semantics of mono, bi, and tri-grid, you might point out that these naming conventions are just conventions for the grid aliaes; i.e., technically for mono-grid even, each component is spelled out in the full grid definition. E.g.,
```
     alias: ne4pg2_ne4pg2 (only for compsets that are (DOCN|XOCN|SOCN|AQP1) )

       longname: a%ne4np4.pg2_l%ne4np4.pg2_oi%ne4np4.pg2_r%r05_g%null_w%null_z%null_m%oQU240
       non-default grids are: atm:ne4np4.pg2  lnd:ne4np4.pg2  ocnice:ne4np4.pg2  rof:r05  glc:null  wav:null  
       mask is: oQU240
```

Reply:
Added a note about this